_studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_vaapi.cpp

```diff
@@ -187,25 +188,39 @@ mfxStatus VAAPIEncoder::CreateAccelerationService(mfxVideoParam const & par)
     }
 
     // Configuration
-    VAConfigAttrib attrib;
+    mfxPriority priority;
+    VAConfigAttrib attrib[2];
 
-    attrib.type = VAConfigAttribRTFormat;
+    attrib[0].type = VAConfigAttribRTFormat;
+    attrib[1].type = VAConfigAttribContextPriority;
 
     vaSts = vaGetConfigAttributes(m_vaDisplay,
                           VAProfileJPEGBaseline,
                           VAEntrypointEncPicture,
-                          &attrib, 1);
+                          &attrib[0], 2);
     MFX_CHECK_WITH_ASSERT(VA_STATUS_SUCCESS == vaSts, MFX_ERR_DEVICE_FAILED);
 
-    if (!(attrib.value & VA_RT_FORMAT_YUV420) || !(attrib.value & VA_RT_FORMAT_YUV422) ) //to be do
+    if (!(attrib[0].value & VA_RT_FORMAT_YUV420) || !(attrib[0].value & VA_RT_FORMAT_YUV422) ) //to be do
         return MFX_ERR_DEVICE_FAILED;
 
+    priority = m_core->GetSession()->m_priority;
+    if (priority == MFX_PRIORITY_LOW)
+    {
+        attrib[1].value = -512;
+    } else if (priority == MFX_PRIORITY_HIGH)
+    {
+        attrib[1].value = 512;
+    } else
+    {
+        attrib[1].value = 0;
+    }
+
     vaSts = vaCreateConfig(
         m_vaDisplay,
         VAProfileJPEGBaseline,
         VAEntrypointEncPicture,
-        NULL,
-        0,
+        &attrib[0],
```

Comment:
I don't like such hardcoding. Could you use ```vector``` instead? then we will have much clear code:

```
vaSts = vaCreateConfig(
        m_vaDisplay,
        VAProfileJPEGBaseline,
        VAEntrypointEncPicture,
        attrib.data(),
        attrib.size(),
        &m_vaConfig);
```

Reply:
+1

Reply:
Just inherit from the previous implementation. Anyway, i change to use std::vector.